File: Module5/Case/FuramaNew/src/app/models/rent-type.ts
export class RentType implements IRentType{
  rentTypeCost: number;
  rentTypeId: number;
  rentTypeName: string;

  constructor(rentTypeCost: number, rentTypeId: number, rentTypeName: string) {
    this.rentTypeCost = rentTypeCost;
    this.rentTypeId = rentTypeId;
    this.rentTypeName = rentTypeName;
  }
}

export interface IRentType {
  rentTypeId: number;
  rentTypeName: string;
  rentTypeCost: number;
}

File: Module5/Case/RestFurama/src/main/java/thanh/code/models/RentType.java
package thanh.code.models;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import java.util.Set;

@Entity
public class RentType {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int rentTypeId;
    @NotBlank
    private String rentTypeName;
    private double rentTypeCost;

    @OneToMany(mappedBy = "rentType", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JsonBackReference(value = "service")
    private Set<Service> service;

    public Set<Service> getService() {
        return service;
    }

    public void setService(Set<Service> service) {
        this.service = service;
    }

    public RentType() {
    }

    public int getRentTypeId() {
        return rentTypeId;
    }

    public void setRentTypeId(int rentTypeId) {
        this.rentTypeId = rentTypeId;
    }

    public String getRentTypeName() {
        return rentTypeName;
    }

    public void setRentTypeName(String rentTypeName) {
        this.rentTypeName = rentTypeName;
    }

    public double getRentTypeCost() {
        return rentTypeCost;
    }

    public void setRentTypeCost(double rentTypeCost) {
        this.rentTypeCost = rentTypeCost;
    }
}
